src/UI/Art/SpriteArtist/SpriteUIUtils.java

```diff
@@ -181,6 +205,45 @@ public static void drawMenuFrame(Graphics g, Color bg, Color frame, int x, int y
     g.fillRect(x, y + h - menuFrameHeight, w, menuFrameHeight); // Lower frame;
   }
 
+  /**
+   * Draws itself as a two-tone box with an image on top.
+   */
+  public static class ImageFrame
+  {
+    public final int xPos;
+    public final int yPos;
+    public final int width;
+    public final int height;
+    private Color mainColor;
+    private Color rimColor;
+    private boolean rimIsUp;
+    private BufferedImage content;
+    public ImageFrame(int x, int y, int w, int h, Color main, Color rim, boolean rimUp, BufferedImage display)
+    {
+      xPos = x;
+      yPos = y;
+      width = w;
+      height = h;
+      mainColor = main;
+      rimColor = rim;
+      rimIsUp = rimUp;
+      content = SpriteLibrary.createDefaultBlankSprite(width, height);
+      Graphics graphics = content.getGraphics();
+      graphics.setColor(rimColor);
+      graphics.fillRect(0, 0, width, height);
+      int dx = 0, dy = 0;
+      if( rimIsUp ) dy++; else dx++;
```

Comment:
So, this produces *either* a left border *or* a top border, but not both? That's... just an interesting design choice.

Reply:
Left and bottom, or top and right, yes. Lets me put a lighter top rim on dark panels and a darker bottom rim on lighter panels. Gives em that nice slightly-raised look.